In [1]:
import pandas as pd

In [2]:
df= pd.read_csv("data/NFHS_FEATURED_DATA.csv").drop(columns= "Unnamed: 0")

In [3]:
df.head()

,Unique_ID,Sex of household member,State,Type of place of residence,Wealth index combined,District number,Age of household members,Combined_BMI,BMI_Category,Obesity,Age_Group
0,0100101305_4,2.0,1.0,2.0,3.0,1.0,22.0,17.34,Underweight,0,21-30
1,0100101305_5,2.0,1.0,2.0,3.0,1.0,19.0,15.86,Underweight,0,15-20
2,0100101345_2,2.0,1.0,2.0,3.0,1.0,38.0,34.49,Obese,1,31-40
3,0100101383_2,2.0,1.0,2.0,3.0,1.0,40.0,26.73,Overweight,0,31-40
4,0100101383_3,2.0,1.0,2.0,3.0,1.0,22.0,15.33,Underweight,0,21-30


In [4]:
df.describe()

,Sex of household member,State,Type of place of residence,Wealth index combined,District number,Age of household members,Combined_BMI,Obesity
count,807417.000000,807417.000000,807417.000000,807417.000000,807417.000000,807417.000000,807417.000000,807417.000000
mean,1.878544,17.171760,1.755893,2.885438,408.751228,30.590909,22.301286,0.051187
std,0.326657,9.604555,0.429557,1.376160,273.619126,10.039606,4.263164,0.220378
min,1.000000,1.000000,1.000000,1.000000,1.000000,15.000000,12.000000,0.000000
25%,2.000000,9.000000,2.000000,2.000000,180.000000,22.000000,19.340000,0.000000
50%,2.000000,18.000000,2.000000,3.000000,367.000000,30.000000,21.730000,0.000000
75%,2.000000,24.000000,2.000000,4.000000,582.000000,39.000000,24.520000,0.000000
max,2.000000,37.000000,2.000000,5.000000,932.000000,98.000000,59.990000,1.000000


In [5]:
pd.crosstab(df["Wealth index combined"],df["Obesity"])

Obesity,0,1
Wealth index combined,,
1.0,165180,2250
2.0,175544,4831
3.0,162408,7900
4.0,144613,11259
5.0,118343,15089


In [6]:
df["BMI_Category"].value_counts()

BMI_Category
Normal         488717
Underweight    140913
Overweight     136458
Obese           41329
Name: count, dtype: int64

## STATISTICAL TESTS

In [7]:
from scipy.stats import chi2_contingency

table = pd.crosstab(df["Sex of household member"],df["Obesity"])
table_percent = pd.crosstab(
    df['Sex of household member'],
    df['Obesity'],
    normalize='index'
) * 100
chi2,p,dof,expected = chi2_contingency(table)
print(table_percent)
print(p)

Obesity                          0         1
Sex of household member                     
1.0                      96.348378  3.651622
2.0                      94.678516  5.321484
1.641023715802988e-109


Outcome: A Chi-square test of independence was conducted to examine the association between sex and obesity status. The association was statistically significant (p < 0.001). Among males, 3.65% were obese compared to 5.32% of females, indicating a significantly higher prevalence of obesity among female respondents.

In [9]:
f_df=df[df["Sex of household member"]==2.0]
m_df=df[df["Sex of household member"]==1.0]

In [10]:
# RURAL VS URBAN
## fOR FEMALE

table = pd.crosstab(f_df["Type of place of residence"],f_df["Obesity"])
table_percent = pd.crosstab(
    f_df['Type of place of residence'],
    f_df['Obesity'],
    normalize='index'
) * 100
chi2,p,dof,expected = chi2_contingency(table)
print(table_percent)
print(p)

Obesity                             0         1
Type of place of residence                     
1.0                         90.492391  9.507609
2.0                         96.020927  3.979073
0.0


In [11]:
## fOR MALE

table = pd.crosstab(m_df["Type of place of residence"],m_df["Obesity"])
table_percent = pd.crosstab(
    m_df['Type of place of residence'],
    m_df['Obesity'],
    normalize='index'
) * 100
chi2,p,dof,expected = chi2_contingency(table)
print(table_percent)
print(p)

Obesity                             0         1
Type of place of residence                     
1.0                         94.133977  5.866023
2.0                         97.100163  2.899837
9.77598700937979e-103


In [12]:
pd.crosstab(
    [df['Sex of household member'],
     df['Type of place of residence']],
    df['Obesity'],
    normalize='index'
)*100

Obesity                                                     0         1
Sex of household member Type of place of residence                     
1.0                     1.0                         94.133977  5.866023
                        2.0                         97.100163  2.899837
2.0                     1.0                         90.492391  9.507609
                        2.0                         96.020927  3.979073

Does obesity become significantly more common as wealth increases, and does this pattern differ between urban and rural populations?

In [13]:
df['Type of place of residence'].value_counts()

Type of place of residence
2.0    610321
1.0    197096
Name: count, dtype: int64

In [14]:
pd.crosstab(
    [df['Type of place of residence'],
     df['Wealth index combined']],
    df['Obesity'],
    normalize='index'
)*100

Obesity                                                   0          1
Type of place of residence Wealth index combined                      
1.0                        1.0                    97.613086   2.386914
                           2.0                    95.984065   4.015935
                           3.0                    93.880742   6.119258
                           4.0                    91.499823   8.500177
                           5.0                    87.911794  12.088206
2.0                        1.0                    98.694863   1.305137
                           2.0                    97.448001   2.551999
                           3.0                    95.729209   4.270791
                           4.0                    93.562730   6.437270
                           5.0                    89.945280  10.054720

In [19]:
df.head()

,Unique_ID,Sex of household member,State,Type of place of residence,Wealth index combined,District number,Age of household members,Combined_BMI,BMI_Category,Obesity,Age_Group
0,0100101305_4,2.0,1.0,2.0,3.0,1.0,22.0,17.34,Underweight,0,21-30
1,0100101305_5,2.0,1.0,2.0,3.0,1.0,19.0,15.86,Underweight,0,15-20
2,0100101345_2,2.0,1.0,2.0,3.0,1.0,38.0,34.49,Obese,1,31-40
3,0100101383_2,2.0,1.0,2.0,3.0,1.0,40.0,26.73,Overweight,0,31-40
4,0100101383_3,2.0,1.0,2.0,3.0,1.0,22.0,15.33,Underweight,0,21-30


In [17]:
ob_df=df[df['Obesity']==1]
pd.crosstab(
        ob_df['Type of place of residence'],
        ob_df['Wealth index combined'],
    normalize='index'
)*100

Wealth index combined,1.0,2.0,3.0,4.0,5.0
Type of place of residence,,,,,
1.0,0.801839,3.504542,11.629472,28.305484,55.758663
2.0,8.967866,17.901681,24.796765,26.435412,21.898276


In [18]:
nob_df=df[~(df['Obesity']==1)]
pd.crosstab(
        nob_df['Type of place of residence'],
        nob_df['Wealth index combined'],
    normalize='index'
)*100

Wealth index combined,1.0,2.0,3.0,4.0,5.0
Type of place of residence,,,,,
1.0,3.262264,8.333054,17.749997,30.312615,40.342069
2.0,27.151490,27.368590,22.253445,15.383436,7.843040


In [41]:
# pd.crosstab(
#     [df['Type of place of residence'],
#      df['Wealth index combined']],
#     df['Obesity'],
#     normalize='index'
# )*100
table = pd.crosstab(
    [df['Type of place of residence'],
     df['Wealth index combined']],
    df['Obesity']
)


print(table)

Obesity                                                0     1
Type of place of residence Wealth index combined              
1.0                        1.0                      5848   143
                           2.0                     14938   625
                           3.0                     31819  2074
                           4.0                     54339  5048
                           5.0                     72318  9944
2.0                        1.0                    159332  2107
                           2.0                    160606  4206
                           3.0                    130589  5826
                           4.0                     90274  6211
                           5.0                     46025  5145


Outcome: Descriptive analysis demonstrated a clear positive socioeconomic gradient in obesity. In both urban and rural populations, obesity prevalence increased progressively from the poorest to the richest wealth quintiles. Across nearly all wealth categories, urban residents exhibited a higher prevalence of obesity than rural residents.

#### Chi-Square Test

In [36]:
urban = df[df['Type of place of residence']==1]

table = pd.crosstab(
    urban['Wealth index combined'],
    urban['Obesity']
)

chi2_contingency(table)

Chi2ContingencyResult(statistic=np.float64(2100.6709921082856), pvalue=np.float64(0.0), dof=4, expected_freq=array([[ 5448.91140358,   542.08859642],
       [14154.80022933,  1408.19977067],
       [30826.23171449,  3066.76828551],
       [54013.43707635,  5373.56292365],
       [74818.61957625,  7443.38042375]]))

In [37]:
rural = df[df['Type of place of residence']==2]

table = pd.crosstab(
    rural['Wealth index combined'],
    rural['Obesity']
)

chi2_contingency(table)

Chi2ContingencyResult(statistic=np.float64(10707.24851651065), pvalue=np.float64(0.0), dof=4, expected_freq=array([[155224.2223584 ,   6214.7776416 ],
       [158467.37489288,   6344.62510712],
       [131163.54965666,   5251.45034334],
       [ 92770.70035277,   3714.29964723],
       [ 49200.1527393 ,   1969.8472607 ]]))

Outcome: Chi-square tests revealed a statistically significant association between household wealth and obesity in both urban (χ² = 2100.67, df = 4, p < 0.001) and rural populations (χ² = 10707.25, df = 4, p < 0.001). These findings indicate that obesity prevalence varies significantly across wealth quintiles in both residential settings.

In [16]:
# result = (
#     df.groupby(
#         ['Type of place of residence',
#          'Wealth index combined',
#          'Obesity']
#     )
#     .size()
#     .groupby(level=0)
#     .apply(lambda x: x/x.sum()*100)
# )

# print(result)

#### LOGISTIC RGRESSION

In [42]:
import statsmodels.formula.api as smf

model = smf.logit(
    "Obesity ~ \
    C(Q('Wealth index combined')) + \
    C(Q('Type of place of residence')) + \
    C(Q('Sex of household member')) + \
    Q('Age of household members')",
    data=df
).fit()

print(model.summary())

Optimization terminated successfully.
         Current function value: 0.181864
         Iterations 8
                           Logit Regression Results                           
Dep. Variable:                Obesity   No. Observations:               807417
Model:                          Logit   Df Residuals:                   807409
Method:                           MLE   Df Model:                            7
Date:                Tue, 07 Jul 2026   Pseudo R-squ.:                 0.09966
Time:                        21:41:31   Log-Likelihood:            -1.4684e+05
converged:                       True   LL-Null:                   -1.6309e+05
Covariance Type:            nonrobust   LLR p-value:                     0.000
                                                coef    std err          z      P>|z|      [0.025      0.975]
-------------------------------------------------------------------------------------------------------------
Intercept                                    -

In [43]:
import numpy as np
import pandas as pd

or_table = pd.DataFrame({
    'Odds Ratio': np.exp(model.params),
    'Lower CI': np.exp(model.conf_int()[0]),
    'Upper CI': np.exp(model.conf_int()[1]),
    'P value': model.pvalues
})

print(or_table)

                                           Odds Ratio  Lower CI  Upper CI  \
Intercept                                    0.001570  0.001463  0.001685   
C(Q('Wealth index combined'))[T.2.0]         1.986164  1.888040  2.089388   
C(Q('Wealth index combined'))[T.3.0]         3.337984  3.182396  3.501179   
C(Q('Wealth index combined'))[T.4.0]         5.029873  4.799462  5.271345   
C(Q('Wealth index combined'))[T.5.0]         7.493522  7.146846  7.857014   
C(Q('Type of place of residence'))[T.2.0]    0.750212  0.733150  0.767671   
C(Q('Sex of household member'))[T.2.0]       1.773849  1.711407  1.838569   
Q('Age of household members')                1.060147  1.059026  1.061268   

                                                 P value  
Intercept                                   0.000000e+00  
C(Q('Wealth index combined'))[T.2.0]       2.920957e-155  
C(Q('Wealth index combined'))[T.3.0]        0.000000e+00  
C(Q('Wealth index combined'))[T.4.0]        0.000000e+00  
C(Q('Wealth

Does wealth affect obesity differently in urban and rural areas?

In [44]:
interaction_model = smf.logit(
    "Obesity ~ \
    C(Q('Wealth index combined')) * C(Q('Type of place of residence')) + \
    C(Q('Sex of household member')) + \
    Q('Age of household members')",
    data=df
).fit()

print(interaction_model.summary()) 

Optimization terminated successfully.
         Current function value: 0.181813
         Iterations 8
                           Logit Regression Results                           
Dep. Variable:                Obesity   No. Observations:               807417
Model:                          Logit   Df Residuals:                   807405
Method:                           MLE   Df Model:                           11
Date:                Tue, 07 Jul 2026   Pseudo R-squ.:                 0.09991
Time:                        21:47:29   Log-Likelihood:            -1.4680e+05
converged:                       True   LL-Null:                   -1.6309e+05
Covariance Type:            nonrobust   LLR p-value:                     0.000
                                                                                     coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------------------------------------------

In [45]:
odd_table = pd.DataFrame({
    'Odds Ratio': np.exp(interaction_model.params),
    'Lower CI': np.exp(interaction_model.conf_int()[0]),
    'Upper CI': np.exp(interaction_model.conf_int()[1]),
    'P value': interaction_model.pvalues
})

print(odd_table)

                                                    Odds Ratio  Lower CI  \
Intercept                                             0.002166  0.001819   
C(Q('Wealth index combined'))[T.2.0]                  1.707893  1.419303   
C(Q('Wealth index combined'))[T.3.0]                  2.612214  2.198196   
C(Q('Wealth index combined'))[T.4.0]                  3.658976  3.089524   
C(Q('Wealth index combined'))[T.5.0]                  5.225023  4.417068   
C(Q('Type of place of residence'))[T.2.0]             0.532366  0.448195   
C(Q('Sex of household member'))[T.2.0]                1.775315  1.712826   
C(Q('Wealth index combined'))[T.2.0]:C(Q('Type ...    1.157574  0.954861   
C(Q('Wealth index combined'))[T.3.0]:C(Q('Type ...    1.267915  1.059248   
C(Q('Wealth index combined'))[T.4.0]:C(Q('Type ...    1.393485  1.168027   
C(Q('Wealth index combined'))[T.5.0]:C(Q('Type ...    1.567786  1.314939   
Q('Age of household members')                         1.060204  1.059083   

           

In [46]:
interaction_model1 = smf.logit(
    "Obesity ~ \
    C(Q('Sex of household member')) * C(Q('Type of place of residence')) + \
    C(Q('Wealth index combined')) + \
    Q('Age of household members')",
    data=df
).fit()

print(interaction_model1.summary())

Optimization terminated successfully.
         Current function value: 0.181853
         Iterations 8
                           Logit Regression Results                           
Dep. Variable:                Obesity   No. Observations:               807417
Model:                          Logit   Df Residuals:                   807408
Method:                           MLE   Df Model:                            8
Date:                Tue, 07 Jul 2026   Pseudo R-squ.:                 0.09971
Time:                        21:53:52   Log-Likelihood:            -1.4683e+05
converged:                       True   LL-Null:                   -1.6309e+05
Covariance Type:            nonrobust   LLR p-value:                     0.000
                                                                                       coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------